### PR TITLE
Cosmetic cleaning of uState.ml: a bit of doc, more unity in naming

### PR DESCRIPTION
**Pull request by herbelin**
**Kind:** small API documentation / ML code cleanup

This PR does a small "cleaning" around `uState.ml`, mainly for name uniformization, especially regarding the overloading of the variable name `ctx`. In particular, it attempts to uniformly use `uctx` for elements of type `UState.t`.

----
As an indication, here is a table of how variable names are used in the context of universes handling, together with a suggestion for a more standardized name, easier to map to a type (only `ctx` for `UState.t` is related to the PR):

|       | UState.t | ContextSet.t | universes_ entry    | UGraph.t | LSet.t | Instance.t | AUContext.t | UContext.t |
| ------|:--------:|:------------:|:-------------------:|:--------:|:------:|:----------:|:-----------:|:----------:|
| uctx  |    _X_   |       X      |                     |          |        |            |             |      X     |
| ctx   |    X     |       X      |                     |          |    X   |            |      X      |      X     |
| univs |    X     |       X      |  yes (and as label) |    X     |   X    |      X     |             |            |
| universes |      |              |                     |    X     |        |            |             |            |
| utyp  |          |              |  yes                |          |        |            |             |            |
| suggestion | uctx | ulcxt/lctx ("l" for local) |  uentry/utyp/univs  |  ugraph/univs  |  levs  | inst/uinst | auctx (?)   |  inst (?)  |

Similarly, many universes structures refer to `Context`: here is a possible more discriminating naming if ever some interest arises:
- `ContextSet.t` -> `ConstrainedUnivLevels.t`
- `UContext.t` -> `ConstrainedUnivInstance.t`

An other suggestion for possibly more explicit names:
- `LMap` -> `UnivLevelMap` ?
- `LSet` -> `UnivLevelSet` ?

PS1: I hope noone is doing similar job, otherwise we can of course join forces.

PS2: If no typo is introduced, there should be no semantic changes.

----
More questions:
- Is `UState.add_global_univ` really needed; if I skipped it in `pretyping.ml` at the time of interpreting a known global universe, it fails in `Function`, as if it was necessary to refer to global universes associated to the current definition which have not been added in the global environment or something like that. Is that possible?

More changes:
- move of `interp_univ_decl` to `constrintern.ml`

### CODE REVIEW

**Review by ppedrot — APPROVED** [review]
LGTM.

**Comment by ppedrot** [comment]
Personally I don't really care much about the names of the variables which always end up confusing me, since I rely instead on the type information provided by ocamllsp. But it can't hurt to make things more uniform indeed.

**Comment by herbelin** [comment]
Thanks for the comments, what gives ocamllsp, compared to merlin?

In my practice, uniform names are also helpful when you write new code and wonder what is the best names you should use for such notion.

File: engine/uState.mli [hunk 1]
```diff
@@ -23,25 +23,34 @@ type t
 
 (** {5 Constructors} *)
 
+(** Different ways to create a new universe state *)
+
 val empty : t
```
Comment (gares) [inline engine/uState.mli:28]: IIRC `empty` is plain wrong. One should always use `from_env`.
While you are at it, you could write "don't use!" here.

Comment (herbelin) [reply engine/uState.mli:28]: Good point.

**Comment by SkySkimmer** [comment]
>As an indication, here is a table of how variable names are used in the context of universes handling, together with a suggestion for a more standardized name, easier to map to a type (only ctx for UState.t is related to the PR):
>contextset: lctx (l for local)

Are contextsets used in a particularly "local" way?

>ucontext: inst 

uctx works fine imo

In general I wouldn't try to match names and types 1-to-1, I would just have
- univs: ustate, univ entry, ugraph
- (a)uctx: contextset, (a)ucontext
- something for lset (probably dependent on how the variable is used, eg "keep" in restrict_context_set), something for instances (uinst?)

>Similarly, many universes structures refer to Context: here is a possible more discriminating naming if ever some interest arises:

I don't have much appetite for these changes, if you want them skip the `Univ` which is already there from being defined in univ.ml.

>Is UState.add_global_univ really needed; if I skipped it in pretyping.ml at the time of interpreting a known global universe, it fails in Function, as if it was necessary to refer to global universes associated to the current definition which have not been added in the global environment or something like that. Is that possible?

Who knows what Function does. "It breaks if we remove it" is "really needed" enough for me.

**Comment by ppedrot** [comment]
> I don't have much appetite for these changes

I agree with @SkySkimmer, this is not Java, we're not using names that take half of the line of code in the Coq source. Also, while the names are fairly badly chosen, I grew accustomed to them.

**Comment by ppedrot** [comment]
Merging this soon, last round for comments.

**Pull request merged**